src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,675 @@
+
+        if ($openApiResponseTransfer->getMessages()->count() === 0) {
+            $messageTransfer = new MessageTransfer();
+            $messageTransfer->setMessage('Something went wrong. Either not channels have been found or the channels do not have messages defined.');
+            $openApiResponseTransfer->addError($messageTransfer);
+        }
+
+        return $openApiResponseTransfer;
+    }
+
+    /**
+     * @param string $openApiFilePath
+     *
+     * @return \cebe\openapi\spec\OpenApi
+     */
+    public function load(string $openApiFilePath): OpenApi
+    {
+        return Reader::readFromYamlFile((string)realpath($openApiFilePath));
+    }
+
+    /**
+     * @param string $resourceUri
+     *
+     * @return string
+     */
+    protected function getModuleName(string $resourceUri): string
+    {
+        $resources = explode('/', trim($resourceUri, '/'));
+
+        return ucwords(current($resources)) . 'Api';
+    }
+
+    /**
+     * @param string $resourceUri
+     *
+     * @return string
+     */
+    protected function getController(string $resourceUri): string
+    {
+        $controllerName = static::APPSRESOURCE;
+
+        $resources = explode('/', trim($resourceUri, '/'));
+
+        $lastResource = end($resources);
+
+        if (count($resources) > 1 && is_string($lastResource) && strpos($lastResource, '{') === false) {
+            $controllerName = $this->getInflectorFactory()->classify($lastResource);
+        }
+
+        return "{$controllerName}Controller";
+    }
+
+    /**
+     * @return \Doctrine\Inflector\Inflector
+     */
+    protected function getInflectorFactory(): Inflector
+    {
+        return InflectorFactory::create()->build();
+    }
+
+    /**
+     * @param \cebe\openapi\spec\OpenApi $openApi
+     * @param \Generated\Shared\Transfer\OpenApiResponseTransfer $openApiResponseTransfer
+     *
+     * @return \Generated\Shared\Transfer\OpenApiResponseTransfer
+     */
+    protected function buildCodeForOpenApi(
+        OpenApi $openApi,
+        OpenApiResponseTransfer $openApiResponseTransfer
+    ): OpenApiResponseTransfer {
+        /** @var \cebe\openapi\spec\PathItem $pathItem */
+        foreach ($openApi->paths->getPaths() as $path => $pathItem) {
+            //Parse data api path wise
+            $this->parseApiPathItem($path, $pathItem);
+        }
+
+        //Generating Commands using parsed data
+        $this->generateCommands();
+
+        return $openApiResponseTransfer;
+    }
+
+    /**
+     * @param string $path
+     * @param \cebe\openapi\spec\PathItem $pathItem
+     *
+     * @return void
+     */
+    protected function parseApiPathItem(string $path, PathItem $pathItem): void
+    {
+        /** @var \cebe\openapi\spec\Operation $operation */
+        foreach ($pathItem->getOperations() as $method => $operation) {
+            //Set Module Name
+            $this->parsedData[$path][$method]['moduleName'] = $this->getModuleName($path);
```

Comment:
Don't use a member variable for this its not a good design, hard to understand when it's filled and hard to debug.